airbyte-integrations/connectors/source-exchange-rate-api/source_exchange_rate_api/source.py

```diff
@@ -0,0 +1,154 @@
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        params = {"base": self.config["base"]}
+        if "symbols" in self.config and self.config["symbols"]:
+            params["symbols"] = self.config["symbols"]
+        return params
+
+    @property
+    def state(self) -> Mapping[str, Any]:
+        return {self.cursor_field: self._cursor_value or self._to_cursor_normalized_value(self.config["start_date"])}
+
+    @state.setter
+    def state(self, value: Mapping[str, Any]) -> None:
+        self._cursor_value = self._to_cursor_normalized_value(value[self.cursor_field]) + datetime.timedelta(days=1)
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        return {self.cursor_field: self._to_cursor_normalized_value(latest_record[self.cursor_field])}
+
+    def read_records(
+            self,
+            sync_mode: SyncMode,
+            cursor_field: List[str] = None,
+            stream_slice: Mapping[str, Any] = None,
+            stream_state: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping[str, Any]]:
+        if not stream_slice:
+            return []
+        records = super(HistoricalRates, self).read_records(
+            sync_mode,
+            cursor_field=cursor_field,
+            stream_slice=stream_slice,
+            stream_state=stream_state
+        )
+        for record in records:
+            next_cursor_value = self._to_cursor_normalized_value(record[self.cursor_field])
+            self._cursor_value = max(self._cursor_value, next_cursor_value) if self._cursor_value else next_cursor_value
+            yield record
+
+    def _chunk_date_range(self, start_date: datetime.date) -> List[Mapping[str, Any]]:
+        dates = []
+        while start_date <= datetime.date.today():
+            dates.append({self.cursor_field: start_date.strftime(self._date_format)})
+            start_date += datetime.timedelta(days=1)
+        return dates or [None]
+
+    def stream_slices(self, stream_state: Mapping[str, Any] = None, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        return self._chunk_date_range(self.state[self.cursor_field])
+
+
+class SourceExchangeRateApi(AbstractSource):
+    @staticmethod
+    def get_authenticator(config: Mapping[str, Any]) -> requests.auth.AuthBase:
+        return HTTPApiKeyHeaderAuth(
+            api_key_header='apikey',
+            access_key=config["access_key"]
+        )
+
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        auth = self.get_authenticator(config)
+        stream = LatestRates(authenticator=auth, config=config)
+        try:
+            next(iter(stream.read_records(sync_mode=SyncMode.full_refresh)))
+        except requests.exceptions.RequestException as e:
+            return False, e
```

Comment:
I'm not  sure if the exception instance would work here. It's better to use its string representation